File: languages/java/src/main/java/com/bloock/sdk/entity/identity/BooleanAttributeDescriptor.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.IdentityEntities;
import com.bloock.sdk.bridge.proto.IdentityEntities.BooleanAttributeDefinition;

/**
 * Represents a descriptor for a boolean attribute.
 */
public class BooleanAttributeDescriptor extends AttributeDescriptor {
  /**
   * Creates a new BooleanAttributeDescriptor instance with the provided details.
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   */
  public BooleanAttributeDescriptor(
      String displayName, String technicalName, String description, Boolean required) {
    super(displayName, technicalName, description, required);
  }

  public static BooleanAttributeDescriptor fromProto(BooleanAttributeDefinition res) {
    return new BooleanAttributeDescriptor(
        res.getDisplayName(), res.getId(), res.getDescription(), res.getRequired());
  }

  public BooleanAttributeDefinition toProto() {
    return IdentityEntities.BooleanAttributeDefinition.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .build();
  }
}

File: languages/go/entity/identityV2/boolean_attribute_descriptor.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type BooleanAttributeDescriptor struct {
	AttributeDescriptor
}

func NewBooleanAttributeDescriptor(name string, id string, description string, required bool) BooleanAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return BooleanAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewBooleanAttributeDescriptorFromProto(s *proto.BooleanAttributeDefinitionV2) BooleanAttributeDescriptor {
	if s == nil {
		return BooleanAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return BooleanAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s BooleanAttributeDescriptor) ToProto() *proto.BooleanAttributeDefinitionV2 {
	return &proto.BooleanAttributeDefinitionV2{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}
